From 5beb99a2cbc929bd06f7bb2fb351391e7f5c1fef Mon Sep 17 00:00:00 2001
From: Vivek Goyal <vgoyal@redhat.com>
Date: Fri, 5 Sep 2014 18:55:45 -0400
Subject: [kernel] kexec: remove unnecessary return

Message-id: <1409943373-12669-11-git-send-email-vgoyal@redhat.com>
Patchwork-id: 91512
O-Subject: [RHEL7.1 PATCH 10/38] kexec: remove unnecessary return
Bugzilla: 852066
RH-Acked-by: Dave Anderson <anderson@redhat.com>
RH-Acked-by: Baoquan He <bhe@redhat.com>
RH-Acked-by: Dave Young <dyoung@redhat.com>
RH-Acked-by: Chao WANG <chaowang@redhat.com>

BZ:852066

commit 80c74f6a40284c5c5d49f3b3289172bbce0b30b8
Author: Xishi Qiu <qiuxishi@huawei.com>
Date: Wed, 11 Sep 2013 14:24:47 -0700

Code can not run here forever, so remove the unnecessary return.

Signed-off-by: Xishi Qiu <qiuxishi@huawei.com>
Suggested-by: Zhang Yanfei <zhangyanfei@cn.fujitsu.com>
Reviewed-by: Simon Horman <horms@verge.net.au>
Reviewed-by: Zhang Yanfei <zhangyanfei@cn.fujitsu.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/kexec.c b/kernel/kexec.c
index f44e7f9..e89d3e5 100644
--- a/kernel/kexec.c
+++ b/kernel/kexec.c
@@ -1535,11 +1535,8 @@ static int __init __parse_crashkernel(char *cmdline,
  if (first_colon && (!first_space || first_colon < first_space))
   return parse_crashkernel_mem(ck_cmdline, system_ram,
     crash_size, crash_base);
- else
-  return parse_crashkernel_simple(ck_cmdline, crash_size,
-    crash_base);
 
- return 0;
+ return parse_crashkernel_simple(ck_cmdline, crash_size, crash_base);
 }
 
 /*
-- 
1.7.1